src/main/java/vazkii/patchouli/client/book/gui/BookTextRenderer.java

```diff
@@ -54,4 +54,8 @@ public void render(int mouseX, int mouseY) {
 	public void click(int mouseX, int mouseY, int mouseButton) {
 		words.forEach(word -> word.click(mouseX, mouseY, mouseButton));
 	}
+
+	public boolean shouldShowTooltip(int mouseX, int mouseY) {
+		return words.stream().anyMatch(word -> !word.hasTooltip() && word.isHovered(mouseX, mouseY)); 
```

Comment:
Do you mean `word -> word.hasTooltip() && word.isHovered(mouseX, mouseY))`, without the `!`?

Reply:
No, it's checking to see if the word already has a tooltip. It makes it cancel showing the tooltip if you're hovering over a link.